public/app/plugins/datasource/kairosdb/datasource.js

```diff
@@ -331,7 +331,8 @@ function (angular, _, dateMath, kbn) {
       }
 
       if (target.tags) {
-        query.tags = angular.copy(target.tags);
+        query.tags = {};
+        angular.copy(target.tags, query.tags);
```

Comment:
this code does not look right, should be able to replace to angular copy and the for each loop with a `_.reduce`  call